classes/article/SubmissionFileDAO.inc.php

```diff
@@ -64,6 +64,8 @@ function getGenreCategoryMapping() {
 	function fromRow($row) {
 		if (isset($row['artwork_file_id']) && is_numeric($row['artwork_file_id'])) {
 			return parent::fromRow($row, 'SubmissionArtworkFile');
+		} elseif (isset($row['supplementary_file_id']) && is_numeric($row['supplementary_file_id'])) {
```

Comment:
Hum, I see Dc11SchemaArticleAdapter now, and the how it handle galleys. I
overlooked that the plugin implemented the DcSchema in an article level.

Although the structure is ready to implement the schema in another object
level (see the assoc type parameter), not sure if we can use the general
PKPDc11Schema for galleys, subject doesn't make much sense to me for files.
I just wanted to make sure we avoid duplicating DC schema in the system
code (inside object classes, for example) instead of using what's ready.

2015-04-07 14:11 GMT-03:00 Alec Smecher <notifications@github.com>:

> In classes/article/SubmissionFileDAO.inc.php
> <https://github.com/pkp/ojs/pull/452#discussion_r27898449>:
>
> > @@ -64,6 +64,8 @@ function getGenreCategoryMapping() {
> >  	function fromRow($row) {
> >  		if (isset($row['artwork_file_id']) && is_numeric($row['artwork_file_id'])) {
> >  			return parent::fromRow($row, 'SubmissionArtworkFile');
> > +		} elseif (isset($row['supplementary_file_id']) && is_numeric($row['supplementary_file_id'])) {
>
> Bruno, thinking about this a little more: OJS doesn't provide
> supplementary files as their own OAI records; each OAI record is a single
> article. The OAI data refers to supplementary files as "relation" elements
> but that's all. I think that's the appropriate thing to do for articles,
> until we have a requirement for more. However, the DC metadata will be
> active in other areas for supplementary content, such as Dataverse.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pkp/ojs/pull/452/files#r27898449>.
>